Java/src/main/java/Airport.java

```diff
@@ -1,110 +1,84 @@
-import Planes.experimentalPlane;
+import planes.ExperimentalPlane;
 import models.MilitaryType;
-import Planes.MilitaryPlane;
-import Planes.PassengerPlane;
-import Planes.Plane;
+import planes.MilitaryPlane;
+import planes.PassengerPlane;
+import planes.Plane;
 
 import java.util.*;
-
-// version: 1.1
-// made by Vitali Shulha
-// 4-Jan-2019
+import java.util.stream.Collectors;
 
 public class Airport {
     private List<? extends Plane> planes;
 
-
-
-    public List<PassengerPlane> getPasPl() {
-        List<? extends Plane> l = this.planes;
-        List<PassengerPlane> x = new ArrayList<>();
-        for (Plane p : l) {if (p instanceof PassengerPlane) {x.add((PassengerPlane) p);}}
-        return x;
+    public Airport(List<? extends Plane> planes) {
+        this.planes = planes;
     }
 
-    public List<MilitaryPlane> getMilitaryPlanes() {
-        List<MilitaryPlane> militaryPlanes = new ArrayList<>();
-        for (Plane plane : planes) {
-            if (plane instanceof MilitaryPlane) {
-                militaryPlanes.add((MilitaryPlane) plane);
-            } //if
-            else {
-
-            } // else
-        } //for
-        return militaryPlanes;
+    public List<? extends Plane> getPlanes() {
+        return planes;
     }
 
-    public PassengerPlane getPassengerPlaneWithMaxPassengersCapacity() {
-        List<PassengerPlane> passengerPlanes = getPasPl();
-        PassengerPlane planeWithMaxCapacity = passengerPlanes.get(0);
-        for (int i = 0; i < passengerPlanes.size(); i++) {
-            if (passengerPlanes.get(i).getPassengersCapacity() > planeWithMaxCapacity.getPassengersCapacity()) {
-                planeWithMaxCapacity = passengerPlanes.get(i);
-            }
-        }
+    public List<PassengerPlane> getPassengerPlanes() {
+        List<PassengerPlane> passengerPlanes = planes.stream()
```

Comment:
All this getXPlanes methods should be refactored to not copy-paste logic

Reply:
I tried to use generics, but could not use expression instanceof T (error Class or array expected)
```
public <T> List<? extends Plane> getPlanes(List<? extends Plane> planes) {
        List<T> filteredPlanes = this.planes.stream()
                .filter(x -> x instanceof T)
                .map(x -> (T) x).collect(Collectors.toList());
        return planes;
```
So, I don't have ideas how to overcome this(